core/src/test/java/org/testcontainers/dockerclient/DockerInDockerTest.java

```diff
@@ -0,0 +1,37 @@
+package org.testcontainers.dockerclient;
+
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+import org.testcontainers.containers.GenericContainer;
+
+import static org.junit.Assume.assumeTrue;
+import static org.rnorth.visibleassertions.VisibleAssertions.assertEquals;
+import static org.testcontainers.junit.GenericContainerRuleTest.getReaderForContainerPort80;
+
+public class DockerInDockerTest {
+
+    @ClassRule
+    public static TestRule assumption = new TestWatcher() {
+        @Override
+        public Statement apply(Statement base, Description description) {
+            assumeTrue("We're inside a container", DockerClientConfigUtils.IN_A_CONTAINER);
```

Comment:
we could add some IP-specific tests here, so this assumption will protect them from running in non-container mode